devrand.c

```diff
@@ -846,8 +849,10 @@ int FdUnMap(pid_t pid, int fd)
 	link_count--;
 
 	if(ocb->links == 0)
+    {
 		free(ocb);
-
+        ocbs_[globalIndex] = NULL;
```

Comment:
I don't think this is correct, I think it assumes the last fd open is the first to close, but will fail with `fd1 = open(/dev/rand); fd2 = open(/dev/rand); close(fd1)`. Wouldn't the correct approach would be to search for the just-deallocated ocb pointer and zero it out?

```
int i = 0;
int LAST = sizeof(ocbs_)/sizeof(ocbs_[0]);
for(int i = 0; i < LAST; i++) {
  if (ocbs_[i] == ocb) {
     ocbs_[i] = NULL;
     break;
  }
}
```